node/src/components/storage.rs

```diff
@@ -1428,15 +1484,25 @@ impl Storage {
             Some(block_header) => block_header,
             None => return Ok(None),
         };
+        self.validate_block_header_hash(&block_header, block_hash)?;
+        Ok(Some(block_header))
+    }
+
+    /// Validates the block header hash against the expected block hash.
+    fn validate_block_header_hash(
```

Comment:
I'm not sure we should be doing that at all. Question for another ticket. @Fraser999

Reply:
Although it's been split into a new method here, it does already exist in `dev`, so I'd be inclined to leave the PR unchanged here.

I'm not sure it's worth a specific new ticket as this behaviour will necessarily be reviewed when we make some of the upcoming planned changes to storage and data types in the near future.